cedar-policy-validator/src/typecheck.rs

```diff
@@ -673,15 +671,11 @@ impl<'a> SingleEnvTypechecker<'a> {
                 );
                 ans_left.then_typecheck(|ty_expr_left, capability_left| match ty_expr_left.data() {
                     // LHS argument is true, so short circuit the `|| to `True` _without_
-                    // typechecking the RHS. We still need to build an annotated `||` with
-                    // some RHS, so we use a literal `true`.  Contrary to `&&`, we keep a
-                    // capability  when short circuiting `||`. The left operand is `true`,
-                    // so its capability is maintained. The right operand is not evaluated,
-                    // so its capability is not considered.
+                    // typechecking the RHS.
```

Comment:
I think we should still keep the comment about capabilities that begins "Contrary to `&&`" ?